### ISSUE TITLE
Bypasses to equipping weapons restriction.

### ISSUE THREAD

**Issue by jptrrs** [jptrrs/HumanResources#16]
Captain Cookie:
Bug #4: Awesome Inventory's "Find and equip weapon" feature allows pawns to equip weapons they have no proficiency in. This feature can at least be turned off.

**Comment by jptrrs** [comment]
Maey:
It seems with Combat Extended, telling someone to equip a weapon via a Loadout bypasses the right-click prohibition and allows colonists to equip weapons in which they are not proficient.

**Comment by jptrrs** [comment]
Maey:
There is also an issue with LWM Deep Storage's weapons cabinet, where different types of weapons are stacked together in one space. I haven't been able to isolate whether you have to be proficient with the first weapon in the stack, last weapon in the stack, or every weapon in the stack... if it's the former two that would be another way to bypass proficiencies, if it's the latter it would just be annoying.

**Comment by Exidoous** [comment]
Another method:

The (otherwise unmodded) caravan menu permits you to equip weapons for which a pawn lacks proficiency. That pawn will then have the proficiency by the time the caravan ends (if not sooner). I have not tested whether cycling through the weapons in that menu provides all such proficiencies, or only the one the pawn is equipped with when entering a map.

But please leave us at least one such exploit until weapon proficiency training is fully functional :P

**Comment by jptrrs** [comment]
Maybe!!! (evil laughter)

**Comment by jptrrs** [comment]
Version 0.14 Implements an additional check for equipping weapons from inventory, solving the issue with Awesome Inventory and probably other similar mods as well. Also implemented a solution for Simple Sidearms, and LWM Deep Storage was already covered on the last version. Anything else?

**Comment by Exidoous** [comment]
It looks like you entirely disabled the pawn-enters-map weapon proficiency grant. It has funny results when a (quest-provided) mercenary gets downed for some reason. One came with a biocoded chain shotgun, got downed after a social fight, and then refused to pick up the weapon, as evil. Could be PTSD, but I think it's your fault :P

Thinking about it, is there a way to trigger the proficiency grant at pawn creation rather than pawn-enters-map?

**Comment by Exidoous** [comment]
Building off my last comment, such mercenaries will also drop their locked weapons in order to learn a new weapon (I had a standing task to Learn all pawns/all weapons), and then be unable to equip the one they came with. This may be Royalty DLC only.

**Comment by abrenneke** [comment]
I still get pawns that join the colony, and they have weapons equipped that they don't know how to use once they drop. Could be my specific setup, but it is kind of annoying.

**Comment by jptrrs** [comment]
Yeah, that should have been a separate issue, so it slipped my radar... And those quest guys are a PITA to reproduce, because each quest is procedural and there's no way to reproduce them exactly... But I took a look, nonetheless, and I think I know what's wrong. This will require some re-writing.

**Comment by Jee-F** [comment]
I'm here because I figured out I was bypassing the restriction with Combat Extended. imo, weapon proficiencies should be dropped from this mod altogether. That would allow focusing the limited dev time on the best part of the mod: tech research. Otherwise, be made optional asap.

**Comment by jptrrs** [comment]
> I'm here because I figured out I was bypassing the restriction with Combat Extended. 

How?

> imo, weapon proficiencies should be dropped from this mod altogether. That would allow focusing the limited dev time on the best part of the mod: tech research. Otherwise, be made optional asap.

Thanks. But that's not going to happen.

**Comment by Jee-F** [comment]
Via loadout, as mentionned above. Sorry, I didn't mean to say I "figured out" something new. Maybe this isn't the right place to discuss this, but it's tagged "open for suggestion".

To be honest, I didn't even try using the proficiencies. Maybe it's not as bad as I think it is. The point is, Rimworld is great because mods allow players to play the game they want. R&D and weapon proficiencies are two different things and I'd rather not have the latter in my game. Practice dummies feel cheaty to me, not to mention yours are a buggy mess, no offense meant. I'm putting up with this because the research part is so awesome.

What if not having the proficiency still allowed to equip and use weapons but instead triggered the "experiment with unknown weapons" accidents when equipped or used in combat. Increased chances of firearms jamming or fumbles with melee. Also, proficiency shouldn't be binary. Thing is, another mod might come up with a better system and it will either be incompatible with yours or will have to turn yours off somehow. You should make it optional while you're still working on it.

**Comment by jptrrs** [comment]
Ok, the issue with equiping weapons in Combat Extended has been [properly noted](https://github.com/jptrrs/HumanResources/wiki/Compatibility-Notes). Thanks. I might take a look on it when I have a chance, but I won't promise anything because CE is really unfriendly to other mods...
